nox session coverage-3.6 fails with python <= 3.6.3

Repro steps:

```
conda create -n py363-tmp python=3.6.3 --yes
conda activate py363-tmp
pytest tests/test_base_config.py::TestCopy
```

Observe multiple errors like:
```
FAILED tests/test_base_config.py::TestCopy::test_copy[src0-<lambda>0] - TypeError: __copy__() takes 1 positional argument but 2 were given
```

The error is related to this item in Python 3.6.4 changelog (https://docs.python.org/3.6/whatsnew/changelog.html#python-3-6-4-final):
```
bpo-28556: Two minor fixes for typing module: allow shallow copying instances of generic classes, [...]
```

I ran into this because on my MacBook `nox` is re-using `/opt/homebrew/opt/python36/bin/python3.6` which is 3.6.2.

I couldn't find a way to force `nox` to use Python >= 3.6.4 so I hacked around a PR to just skip these tests when this happens (will submit it shortly).

--- a/tests/test_base_config.py
+++ b/tests/test_base_config.py
@@ -523,6 +523,17 @@ class TestCopy:
         assert id(cfg[0]) == id(cp[0])
 
 
+# This test is to ensure 100% coverage with Python <= 3.6.3, even when the
+# above tests are skipped.
+@pytest.mark.skipif(  # type: ignore
+    sys.version_info >= (3, 6, 4), reason="Copy is tested in `TestCopy`"
+)
+def test_enforce_copy_coverage() -> None:
+    cfg = OmegaConf.create([])
+    with raises(TypeError):
+        cfg.copy()
+
+
 def test_not_implemented() -> None:
     with raises(NotImplementedError):
         OmegaConf()

--- a/tests/test_base_config.py
+++ b/tests/test_base_config.py
@@ -1,4 +1,5 @@
 import copy
+import sys
 from enum import Enum
 from typing import Any, Dict, List, Union
 
@@ -477,6 +478,11 @@ def test_open_dict_restore(flag_name: str, ctx: Any) -> None:
     assert not cfg.foo._get_node_flag(flag_name)
 
 
+# There is an issue in Python <= 3.6.3 related to copying instances of generic classes,
+# that makes many of the copy tests fail.
+@pytest.mark.skipif(
+    sys.version_info <= (3, 6, 3), reason="Requires Python 3.6.4 or newer"
+)
 @pytest.mark.parametrize("copy_method", [lambda x: copy.copy(x), lambda x: x.copy()])
 class TestCopy:
     @pytest.mark.parametrize(  # type: ignore

